packages/common/src/ContractInteractor.ts

```diff
@@ -773,11 +803,47 @@ calculateTransactionMaxPossibleGas: result: ${result}
       return gasPriceFromNode
     }
     const gasPriceActual =
-      new BN(gasPriceFromNode)
+      toBN(gasPriceFromNode)
         .muln(this.environment.getGasPriceFactor)
     return gasPriceActual.toString()
   }
 
+  async getFeeHistory (blockCount: number | BigNumber | BN | string, lastBlock: number | BigNumber | BN | string, rewardPercentiles: number[]): Promise<FeeHistoryResult> {
+    // web3 incorrectly converts blockCount type from prefixed hex string to number, hardhat doesn't handle that properly
+    // see https://github.com/ChainSafe/web3.js/pull/4529
+    // once it's fixed and web3 version is updated use:
+    // return await this.web3.eth.getFeeHistory(blockCount, lastBlock, rewardPercentiles)
```

Comment:
Note: that PR was merged and 1.6.1 was released since, can we update?

Reply:
It's merged to their 1.x branch, but wasn't part from 1.6.1 release. They will add it in the future probably.